docs2/site/docs/guides/complexity-analyzer.md

```diff
@@ -0,0 +1,321 @@
+# Complexity Analyzer
+
+## Overview
+
+The Complexity Analyzer in GraphQL.NET is a powerful tool designed to manage the complexity and depth of GraphQL queries.
+It ensures that queries remain within acceptable bounds to prevent excessive load on the server. This documentation will
+guide you through the basic and advanced configuration of the complexity analyzer.
+
+### Key Features
+
+- Fields can define the impact for the execution of that field (e.g., how long it will take to execute the ResolveAsync
+  method) separate from a multiplier applied to the children fields (e.g., how many rows it returns for list fields).
+- Default configuration allows for setting three default values:
+  - Scalar impact (impact to use for scalar fields)
+  - Object impact (impact to use for object fields)
+  - Default child multiplier for list fields (multiplier to use for list fields)
+- Per-field configurable behavior to determine field impact and child impact multipliers; by default it uses the
+  scalar impact for scalar fields, object impact for object fields, and the default child multiplier for list fields.
+- Default behavior considers connection semantics, such as `first`, `last`, and `id` arguments, to adjust the child
```

Comment:
`limit` + `offset` are also a very popular solution for pagination (even though not the best practice). Can they also be added to special handling list? Alternativly, can we expose an additional overload for `WithComplexityImpact` with access to `IResolveFieldContext` so that user can create his own logic based on arguments?

Reply:
The functionality already exists but I didn't add documentation. I'll revise the documentation.

Reply:
Oh, wait. We already have such an overload with `FieldImpactContext`, right?

Reply:
Yes

Reply:
Done

Reply:
The default logic can also be redefined within the options, if desired.

Reply:
Thanks for the example!
We use `limit` in many places, so we will need to apply this logic a lot. Is it possible to make the list of supported pagination arguments extensible? So by default, it will contain `first` and `last`, but the user will be able to add `limit` or whatever if needed.
Alternatively, I think, the user can do that with the schema visitor by looking for fields with a `limit` argument and applying the logic from the example.

Reply:
Just redefine the default behavior within the options.  Copy the original code and make a few tweaks.  No need to use a schema visitor.

I don't know how extensible we want the default delegate.  Every field needs to run the code, and if it needs to search another list to know what arguments to look for, it will impact performance further.

Thoughts?

Reply:
Copy+adjust is also an option. The only drawback: if you fix a bug in the original code, I'll be stuck with buggy code 🙂